### PR TITLE
Feature/add auto documentation

### PR DESCRIPTION
**Pull request by gignsky**
I am requesting a pull into the master branch, but for good reason I think.

The only updates in this pr are twofold,

1st, An update to the readme.md with a little badge that says documentation that contains a hyperlink to the MOST LIKELY URL for the documentation (I had to guess at this since I don't have access to your account, but it should probably work)

2nd, a new github actions workflow has been added to the folder, its rather simple I think but I got it running on some other code, basically it will automatically run 'cargo doc' and then take the generated website and throw it into github pages.

NOTE: SETTINGS MIGHT NEED MODIFICATION on the repo's settings panel make sure that github pages has its source set to "Github Actions" This needs to be done **_PRIOR TO MERGING_**

### CODE REVIEW

**Comment by gignsky** [comment]
> NOTE: SETTINGS MIGHT NEED MODIFICATION on the repo's settings panel make sure that github pages has its source set to "Github Actions" This needs to be done **_PRIOR TO MERGING_**

REMINDER!!

**Review by dotunwrap — COMMENTED** [review]
This is an interesting add. Could be convenient. Can you provide steps for testing so I can test this before it's merged? Or do you have a demo you can provide?

File: .github/workflows/docs.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        uses: Swatinem/rust-cache@v2
+      - name: Setup pages
+        id: pages
+        uses: actions/configure-pages@v4
+      - name: Clean docs folder
+        run: cargo clean --doc
+      - name: Build docs
+        run: cargo doc
+      - name: Add redirect
+        run: echo '<meta http-equiv="refresh" content="0;url=credit_card_convenience/index.html">' > target/doc/index.html
```
Comment (dotunwrap) [inline .github/workflows/docs.yml:31]: What is the significance of `credit_card_convenience/index.html`?

**Comment by dotunwrap** [comment]
This PR has gone stale. If you would like to provide further updates or still want to add this feature, please create a new, up-to-date PR.

**Pull request closed without merging**